paasta_tools/kubernetes/bin/paasta_secrets_sync.py

```diff
@@ -152,9 +159,33 @@ def get_services_to_k8s_namespaces(
                 soa_dir=soa_dir,
             )
             if instances:
-                services_to_k8s_namespaces[service].add(
-                    INSTANCE_TYPE_TO_K8S_NAMESPACE[instance_type]
-                )
+                for instance in instances:
+                    try:
+                        service_instance_config = (
+                            load_kubernetes_service_config_no_cache(
+                                service=instance[0],
+                                instance=instance[1],
+                                cluster=cluster,
+                                soa_dir=soa_dir,
+                            )
+                        )
+                    except NoDeploymentsAvailable:
+                        log.debug(
+                            "No deployments found for %s.%s in cluster %s. Skipping."
+                            % (instance[0], instance[1], cluster)
+                        )
+                        continue
+                    except NoConfigurationForServiceError:
+                        log.error(
+                            f"Could not read kubernetes configuration file for %s.%s in cluster %s"
+                            % (instance[0], instance[1], cluster)
+                        )
+                        continue
```

Comment:
It's probably better to refactor this to use PaastaServiceConfigLoader: instead of

```
            instances = get_service_instance_list(
                service=service,
                instance_type=instance_type,
                cluster=cluster,
                soa_dir=soa_dir,
            )
...
                for instance in instances:
                    try:
                        service_instance_config = (
                            load_kubernetes_service_config_no_cache(
```

do something like

```
config_loader = PaastaServiceConfigLoader(service=service, soa_dir=soa_dir)
for service_instance_config in config_loader.instance_configs(cluster=cluster, instance_type_class=...)
```


(I'm realizing I don't think we actually have a mapping anywhere of "kubernetes" -> KubernetesDeploymentConfig, "tron" -> TronActionConfig, etc., so mapping instance_type to an instance_type_class might require defining that.)